core/src/contracts/stablex_contract.rs

```diff
@@ -219,7 +219,7 @@ impl StableXContract for StableXContractImpl {
                 previous_page_user_offset,
                 page_size,
             );
-            builder.block = block_number;
+            builder.block = block_number.map(BlockId::Number);
```

Comment:
I think you can just `.into()` right?

Reply:
BlockId does implement From<BlockNumber> but I still can't `into`, not sure why.